src/bindings/python/flux/future.py

```diff
@@ -43,6 +44,17 @@ def continuation_callback(c_future, opaque_handle):
             del _THEN_HANDLES[py_future]
 
 
+def get_interruptible(func):
```

Comment:
What do you think about putting this in `util` so that down the road we can easily apply it to other python functions/methods that wrap a get (job.wait.wait_get_status), for subclasses of Future (job.wait.JobWaitFuture), or even for decorating a cmd's main function?

If we do, it might be worth just calling it `interruptible` since it could be applied to more than just `get`.

Reply:
Great idea! I'll implement that here. I was so focused on fixing the specific problem I forgot about all those other functions that wrap `flux_future_get(3)`. (However, note that for `job.wait.wait_get_status()` it specifically calls `future.wait_for()` which is already wrapped, so `@interruptible` would not be necessary there.)